lib/puppet/type/cisco_upgrade.rb

```diff
@@ -101,16 +75,33 @@ def self.title_patterns
   # Attributes #
   ##############
 
+  # Deprecated
   newproperty(:version) do
     desc 'Version of the Cisco image to install on the device.
           Valid values are strings'
     validate do |ver|
-      fail "Version can't be nil or an empty string" if
-        ver == '' || ver.nil? == :true
       valid_chars = 'Version can only have the following
           characters: 0-9, a-z, A-Z, (, ) and .'
       fail "Invalid version string. #{valid_chars}" unless
         (/([0-9a-zA-Z().]*)/.match(ver))[0] == ver
     end
   end # property version
+
+  newproperty(:package) do
+    examples = "\nExample:\nbootflash:nxos.7.0.3.I5.2.bin\n
+      tftp://x.x.x.x/path/to/nxos.7.0.3.I5.2.bin\n
+      usb1:nxos.7.0.3.I5.2.bin"
+    supported = "\nNOTE: Only bootflash:,tftp: and usb are supported."
+    desc "{ackage to install on the device. Format <uri>:<image>.
+          Valid values are string.#{examples}#{supported}"
+    validate do |pkg|
+      fail 'Package should be a string.' unless pkg.is_a?(String)
+      fail 'package must match format <uri>:<image>' unless pkg[/\S+:\S+/]
+    end
+  end # property package
+
+  validate do
+    fail "The property 'version' has been deprecated. Please use 'package'." if
```

Comment:
I have taken out the source_uri parameter. Manifest will fail if 'source_uri' is used.